hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestHRegion.java

```diff
@@ -2268,6 +2269,78 @@ public void testCheckAndMutate_WithFiltersAndTimeRange() throws Throwable {
     assertTrue(region.get(new Get(row).addColumn(FAMILY, Bytes.toBytes("A"))).isEmpty());
   }
 
+  @Test
+  public void testCheckAndMutate_wrongMutationType() throws Throwable {
```

Comment:
Are there also positive test cases? The new one's only seem to catch exceptions?

Reply:
Yes, the new tests I added only catch exceptions, but I think we already have positive test cases for HRegion.checkAndMutate() and HRegion.checkAndRowMutate() originally.